src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -856,6 +858,12 @@ internal bool Save(string filePath, bool isBackup = false, EngineController engi
 
                     Model.FileName = filePath;
                     Model.OnSaved();
+
+                    if (Path.GetExtension(filePath).Equals(".dyn") && saveContext.Equals(SaveContext.SaveAs))
+                    {
+                        DynamoViewModel.Model.ClearCurrentWorkspace();
+                        DynamoViewModel.Model.OpenJsonFileFromPath(saveContent, filePath, false);
+                    }
```

Comment:
What happens for `.dyf` files?

Reply:
testing with dyf's

Reply:
Does `OpenJsonFileFromPath` update `WorkspaceModel` with the current `saveContent`?

Reply:
This looks counterintuitive to clear the current workspace and reopen the json file when one is doing a save-as. Is there a better way to update the workspace model on a save-as?